src/pke/lib/schemerns/rns-multiparty.cpp

```diff
@@ -86,11 +119,44 @@ Ciphertext<DCRTPoly> MultipartyRNS::MultipartyDecryptMain(ConstCiphertext<DCRTPo
 
     s.DropLastElements(diffQl);
 
-    DggType dgg(NOISE_FLOODING::MP_SD);
-    DCRTPoly e(dgg, cv[0].GetParams(), Format::EVALUATION);
+    DCRTPoly noise;
+    if (cryptoParams->GetMultipartyMode() == NOISE_FLOODING_MULTIPARTY) {
+        if (sizeQl < 3) {
+            OPENFHE_THROW(config_error, "sizeQl " + std::to_string(sizeQl) +
+                                            " must be at least 3 in NOISE_FLOODING_MULTIPARTY mode.");
+        }
+        DugType dug;
+        auto params                            = cv[0].GetParams();
+        auto cyclOrder                         = params->GetCyclotomicOrder();
+        std::vector<NativeInteger> moduliFirst = {params->GetParams()[0]->GetModulus()};
+        std::vector<NativeInteger> rootsFirst  = {params->GetParams()[0]->GetRootOfUnity()};
+        auto paramsFirst = std::make_shared<ILDCRTParams<BigInteger>>(cyclOrder, moduliFirst, rootsFirst);
+        std::vector<NativeInteger> moduliAllButFirst(sizeQl - 1);
+        std::vector<NativeInteger> rootsAllButFirst(sizeQl - 1);
+        for (size_t i = 1; i < sizeQl; i++) {
```

Comment:
I suggest adding a new method param similar to `PopLastParam()` Call the new one PopFirstParam(). Then call it here.

Reply:
I tried doing that, but it edits the parameter directly, and I was not sure how to make a copy of the params object.

Reply:
The = operator does not work?

Reply:
It's a shared_ptr, but deferencing it worked. Don't remember why I was having trouble with this earlier, but maybe I was just a bit sleepy. Pushed the changes with `PopFirstParam()`.